updateUserFeatureFlag/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Real-Dev-Squad/feature-flag-backend/database"
+	"github.com/Real-Dev-Squad/feature-flag-backend/models"
+	"github.com/Real-Dev-Squad/feature-flag-backend/utils"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/go-playground/validator/v10"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+}
+
+func processUpdateById(userId string, flagId string, requestBody models.UpdateUserMapping) (*models.FeatureFlagUserMapping, error) {
+
+	db := database.CreateDynamoDB()
+
+	input := &dynamodb.UpdateItemInput{
+		TableName: aws.String(database.GetTableName(utils.FFUM_TABLE_NAME)),
+		Key: map[string]*dynamodb.AttributeValue{
+			"userId": {
+				S: aws.String(userId),
+			},
+			"flagId": {
+				S: aws.String(flagId),
+			},
+		},
+		UpdateExpression: aws.String("set #status = :status, #updatedBy = :updatedBy, #updatedAt = :updatedAt"),
+		ExpressionAttributeNames: map[string]*string{
+			"#status":    aws.String("status"),
+			"#updatedBy": aws.String("updatedBy"),
+			"#updatedAt": aws.String("updatedAt"),
+		},
+		ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+			":status": {
+				S: aws.String(strings.ToUpper(requestBody.Status)),
+			},
+			":updatedBy": {
+				S: aws.String(requestBody.UserId),
+			},
+			":updatedAt": {
+				N: aws.String(strconv.FormatInt(time.Now().Unix(), 10)),
+			},
+		},
+		ConditionExpression: aws.String("attribute_exists(userId)"),
```

Comment:
Adding both checks are not required. Conditional Expression in DynamoDB works by first finding an existing item with primary key(userId and flagId) then, evaluate the Condition Expression against the existing item (or null, if there is no existing item) and finally proceeding with the operation if the Condition Expression evaluates to true.
So it will first find an item with the given userId and flagId and if not found will evaluate to null and the condition will evaluate to false.

Reference - https://www.alexdebrie.com/posts/dynamodb-condition-expressions/#how-to-think-about-condition-expressions

Reply:
Resolving this